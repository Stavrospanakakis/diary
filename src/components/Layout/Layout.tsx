import React from 'react'
import PropTypes from 'prop-types'
import '../../assets/main.css'
import Sidebar from '../Sidebar/Sidebar'
import Navbar from '../Navbar/Navbar'


const Layout: React.FC = props => {
	const  { children, isSidebarVisible } = props
	return (
		<div className='xl:container xl:mx-auto'>
			<Navbar />
			{
				isSidebarVisible ?( <Sidebar />) : (<></>)
			}
			<main>{children}</main>
		</div>
	)
}

Layout.propTypes = {
	isSidebarVisible: Boolean ,
	children: PropTypes.node.isRequired
}

export default Layout