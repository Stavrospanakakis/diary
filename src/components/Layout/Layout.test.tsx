import React from 'react'

import { render } from '@testing-library/react'

import Layout from './Layout'

describe('Layout', () => {
	it('should render', () => {
		const { container } = render(<Layout />)
		expect(container).toMatchSnapshot()
	})
})