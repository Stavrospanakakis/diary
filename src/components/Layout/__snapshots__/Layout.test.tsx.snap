// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render 1`] = `
<div>
  <div
    class="xl:container xl:mx-auto"
  >
    <div
      class="bg-gray-900 sm:flex sm:justify-between sm:items-center sm:px-4 sm:py-3 flex items-center justify-center md:justify-between px-4 py-3 sm:p-0"
    >
      <div
        class=""
      >
        <a
          class="text-white font-bold uppercase"
          href="/"
        >
          Diary of a Web Developer
        </a>
      </div>
      <div
        class="px-2 pt-2 pb-4 sm:flex sm:p-0"
      >
        <a
          class="mt-1 block px-2 py-1 text-white rounded hover:bg-gray-800 sm:mt-0 sm:ml-2 hidden md:flex"
          href="https://github.com"
        >
          Projects
        </a>
        <a
          class="mt-1 block px-2 py-1 text-white rounded hover:bg-gray-800 sm:mt-0 sm:ml-2 hidden md:flex"
          href="https://resume.com"
        >
          Resume
        </a>
      </div>
    </div>
    <div
      class="block md:grid md:grid-cols-3 md:gap-4"
    />
  </div>
</div>
`;
